universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -0,0 +1,48 @@
+package controllers.admin;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static play.mvc.Results.notFound;
+import static play.mvc.Results.ok;
+
+import auth.Authorizers;
+import javax.inject.Inject;
+import org.pac4j.play.java.Secure;
+import play.mvc.Http.Request;
+import play.mvc.Result;
+import services.program.BlockDefinition;
+import services.program.ProgramBlockDefinitionNotFoundException;
+import services.program.ProgramDefinition;
+import services.program.ProgramNotFoundException;
+import services.program.ProgramService;
+import views.admin.programs.ProgramBlockPredicatesEditView;
+
+public class AdminProgramBlockPredicatesController {
+  private final ProgramService programService;
+  private final ProgramBlockPredicatesEditView predicatesEditView;
+
+  @Inject
+  public AdminProgramBlockPredicatesController(
+      ProgramService programService, ProgramBlockPredicatesEditView predicatesEditView) {
+    this.programService = checkNotNull(programService);
+    this.predicatesEditView = checkNotNull(predicatesEditView);
+  }
+
+  @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
+  public Result edit(Request request, long programId, long blockDefinitionId) {
+    try {
+      ProgramDefinition programDefinition = programService.getProgramDefinition(programId);
+      BlockDefinition blockDefinition = programDefinition.getBlockDefinition(blockDefinitionId);
+      return ok(
+          predicatesEditView.render(
+              request,
+              programDefinition,
+              blockDefinition,
+              programDefinition.getAvailablePredicateQuestionDefinitions(blockDefinitionId)));
+    } catch (ProgramNotFoundException | ProgramBlockDefinitionNotFoundException e) {
+      return notFound(e.toString());
```

Comment:
(blocking): I don't think we want to push exception messages to our users

Reply:
Added better messages depending on the Exception.